File: backend/src/models/VendorProductModel.ts
import { Schema, model, models } from "mongoose";

const VendorProductSchema = new Schema({
  vendor: { type: Schema.Types.ObjectId, ref: "Vendor", required: true },
  product: { type: Schema.Types.ObjectId, ref: "Product", required: true },
  rating: { type: Number, required: true },
  pricingRules: [
    { type: Schema.Types.ObjectId, ref: "PricingRule", required: true },
  ],
  deliverySlots: [
    { type: Schema.Types.ObjectId, ref: "DeliverySlot", required: true },
  ],
  quantityPricings: [
    { type: Schema.Types.ObjectId, ref: "QuantityPricing", required: true },
  ],
  pricingRuleMetas: [
    { type: Schema.Types.ObjectId, ref: "PricingRuleMeta", required: false },
  ],
});
// Add toJSON transformation
VendorProductSchema.set("toJSON", {
  transform: function (doc, ret) {
    ret.id = ret._id;
    delete ret._id;
    delete ret.__v;
    return ret;
  },
});

export default models.VendorProduct ||
  model("VendorProduct", VendorProductSchema);

File: price-engine/app/models/vendor_product_model.py
from odmantic import Model  
from typing import List 
from .product_model import Product
from .vendor_model import Vendor
from .pricing_rule import PricingRule
from .delivery_slot_model import DeliverySlot
from .quantity_pricing_model import QuantityPricing
 
class VendorProduct(Model):
 
    vendor: Vendor
    product: Product
    rating: float
    pricingRules: List[PricingRule] = []
    deliverySlots: List[DeliverySlot] = []
    quantityPricings: List[QuantityPricing] = [] 
